### PR TITLE
TLS subscription resource

### PR DESCRIPTION
**Pull request by bengesoff**
Need to add documentation still but implementation seems to work fine

### CODE REVIEW

File: fastly/resource_fastly_tls_certificate_test.go [hunk 1]
```diff
@@ -35,7 +35,7 @@ func TestAccFastlyTLSCertificate_withName(t *testing.T) {
 		CheckDestroy: testAccCheckTLSCertificateDestroy,
 		Steps: []resource.TestStep{
 			{
-				Config: testAccTlsCertificateWithName(name, key, name, cert),
+				Config: testAccTlSCertificateWithName(name, key, name, cert),
```
Comment (wjam) [inline fastly/resource_fastly_tls_certificate_test.go:38]: I presume this is supposed to be `TLS` rather than `TlS`?

Comment (bengesoff) [reply fastly/resource_fastly_tls_certificate_test.go:38]: 😂😂😂 Well that's embarrassing

Comment (bengesoff) [reply fastly/resource_fastly_tls_certificate_test.go:38]: 😂😂😂 Well that's embarrassing

File: fastly/resource_fastly_tls_subscription.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+	return &schema.Resource{
+		Create: resourceFastlyTLSSubscriptionCreate,
+		Read:   resourceFastlyTLSSubscriptionRead,
+		Delete: resourceFastlyTLSSubscriptionDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough, // FIXME: might want to do something with adding a "subscription_validation" resource too
+		},
+		Schema: map[string]*schema.Schema{
+			"domains": {
+				Type:        schema.TypeList,
```
Comment (wjam) [inline fastly/resource_fastly_tls_subscription.go:19]: This should probably include validation that there's at least one item in the list

File: fastly/resource_fastly_tls_subscription.go [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+				Description: "List of domains on which to enable TLS.",
+				Required:    true,
+				ForceNew:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"certificate_authority": {
+				Type:        schema.TypeString,
+				Description: "The entity that issues and certifies the TLS certificates for your subscription. Valid values are `lets-encrypt` or `globalsign`.",
```
Comment (wjam) [inline fastly/resource_fastly_tls_subscription.go:27]: This should validate that the valid values are those correct values

File: fastly/resource_fastly_tls_subscription_test.go [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+func init() {
+	resource.AddTestSweepers("fastly_tls_subscription", &resource.Sweeper{
+		Name: "fastly_tls_subscription",
+		F:    testSweepTLSSubscription,
+	})
+}
+
+func TestAccResourceFastlyTLSSubscription(t *testing.T) {
+	name := acctest.RandomWithPrefix(testResourcePrefix)
+	domain := fmt.Sprintf("%s.com", name)
```
Comment (wjam) [inline fastly/resource_fastly_tls_subscription_test.go:23]: Change the TLD to `example` or `test` to avoid accidentally selecting a real domain

Comment (wjam) [reply fastly/resource_fastly_tls_subscription_test.go:23]: https://en.wikipedia.org/wiki/.example

**Comment by wjam** [comment]
Plus already mentioned website documentation

**Comment by wjam** [comment]
Build still failing

**Comment by bengesoff** [comment]
Fixed build but will have to update again when https://github.com/opencredo/go-fastly/pull/30 merges

**Comment by wjam** [comment]
👍 once the go-fastly commit has been changed to the merged one

**Review by wjam — APPROVED** [review]

File: fastly/resource_fastly_tls_subscription.go [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+				Type:        schema.TypeList,
+				Description: "List of domains on which to enable TLS.",
+				Required:    true,
+				ForceNew:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				MinItems:    1,
+			},
+			"certificate_authority": {
+				Type:         schema.TypeString,
+				Description:  "The entity that issues and certifies the TLS certificates for your subscription. Valid values are `lets-encrypt` or `globalsign`.",
```
Comment (trentrosenbaum) [inline fastly/resource_fastly_tls_subscription.go:29]: I know that last year they were trying to move over to lets-encrypt and phase out globalsign.  If the API says that it accepts both, but will confirm that.

**Review by trentrosenbaum — APPROVED** [review]
Looks good and after this is merged the authorisations/challenges can be added.

File: fastly/resource_fastly_tls_subscription.go (hunk 5 above)
Comment (bengesoff) [reply fastly/resource_fastly_tls_subscription.go:29]: Oh interesting - well it says globalsign is "limited availability" so I guess at least someone could use it 🤷🏻‍♂️

**Pull request merged**